### ISSUE TITLE
Incorrect Path -> Failed Git Error Documentation

### ISSUE THREAD

**Issue by aicayzer** [denolehov/obsidian-git#583]
Reproduction Steps:
1. Create a vault 
2. Change the name of a parent folder to that vault (e.g. /Documents/Vault -> /NewDocuments/Vault)
3. Obsidian Git will say "Can't find a valid git repository. Please create
one via the given command or clone an
existing repo.”

Thankfully I was able to diagnose the issue pretty quickly but on another day, or for those less technically inclined, I can see this being a bigger issue. 

Suggested Outcome:
Add a “Warning” to the Documentation or Readme.

**Comment by Vinzent03** [comment]
Did you move your vault or did you have to add that folder as a new vaul? And what have to done to fix that. Please use the bug template next time!

**Comment by aicayzer** [comment]
Hey,My apologies! Will do as a bug next time. I created the vault & then, outside of Obsidian, renamed the parent folder which is was in. e.g. /Documents/Folder/Vault -> /Documents/SameFolderDifferentName/VaultTo fix it, I removed the Vault from Obsidian & then re-added it. i.e. clearly somewhere Obsidian Git gets the route for the Vault from Obsidian. Although Obsidian continues to work if you rename the parent folder (or any other folder between the vault and the root directory of your device), the update doesn’t perpetuate to Obsidian Git (maybe an Obsidian issue). Let me know if you want any further details :) Sent from my iPhoneOn 23 Aug 2023, at 7:10 pm, Vinzent ***@***.***> wrote:﻿
Did you move your vault or did you have to add that folder as a new vaul? And what have to done to fix that. Please use the bug template next time!

**Comment by Vinzent03** [comment]
Okay so you renamed a parent folder of the vault and therefore had to re-add the vault to Obsidian. Everything should still work in the plugin except settings that are not stored in the vault are lost, but those shouldn't affect this error message. Does the plugin work now and if so have you changed something to fix it? What exactly do you want to be documented?

**Comment by aicayzer** [comment]
Prior to realising the issue, I was very confused why Obsidian Git didn't work. I think it would be good if the scenario I ran into was documented because I can for-see others having this issue in the future. Let me recap just so we're clear:

My Obsidian Vault (despite having renamed the parent folder) worked as normal. i.e. i didn't have to remove & re-add it, it continued to work irrelevant of the renamed parent folder. So too did all my other plugins. 

A month later, I installed the Obsidian Git plug-in. Despite having a working Git repository in the Vault Directory, it kept showing the error message. 

Given Obsidian was working as normal &, 6 months ago I'd been using the Plugin with this Vault (uninstalled in-between), I was very confused why the Plug-in kept saying there wasn't a Git repository configured. 

So what do I want documented/fixed? 

Either an updated to the plug-in which, if Obsidian Git's stored value for the route directory =! the actual route directory, shows an error and says "remove & re-add vault". 

Or an update to the documentation troubleshooting saying "If you're having issues & you've renamed the parent file, try removing the Vault and re-adding it to Obsidian". 

Or a fix the code which leads to this issue. I haven't had time to properly inspect the code but I'll try do so and, if i spot it, will open a PR (if I have the required permissions). 


It's certainly an unusual issue so I understand if it doesn't make the documentation!  


Hopefully that helps & clarifies the issue. 

Thank you
